CONTRIBUTING.md

```diff
@@ -0,0 +1,76 @@
+# Contributing to the Lua Users Foundation
+
+This document serves as contribution guidelines and the points listed should be
+followed for contributions to the foundation.
+
+Contributions are expected to follow the [code of conduct](CODE_OF_CONDUCT.md).
+
+## Commit Guidelines
+
+As is somewhat standard practice, the following should be observed:
+
+- The usage of Markdown-formatted lists is allowed
+- The header line should be no more than 60 characters
+- Commit headers should start with an imperative verb, to act as a description
+of what applying the commit would do: "If applied, this commit will..."
+- Paragraphs should be broken with an empty line between
+- Further lines after the header should be no more than 72 characters
+- Further lines, if required, should contain detailed information on:
+ 1. Implications of the changes in the commit
+ 2. Reasons the changes are required
+
+Commits relating to one file should follow these guidelines:
+
+- The header line should contain the filename and a short description of what
+applying the patch will do, in the format of "If applied, this commit will..."
+- The second line should be blank
+
+Example commit message:
+
+```markdown
+CONTRIBUTING.md: Establish guidelines for new commits
+
+Implications of Changes:
+
+- Could possibly make it harder for outsiders to contribute
+- Could create bloatware in the repository data
+
+Reasons for Changes:
+
+- Creates a structure which can be used to avoid excessive comments
+  during the review process
+```
+
+Commits relating to multiple files should follow these guidelines:
+
+- The header line should contain a short description of what applying the patch
+will do, in the format of "If applied, this commit will..."
+- The second line should be blank
+- The third line should begin a list of changed files, with a header along the
+lines of "Changed Files:"
+
+Example commit message:
+
+```markdown
+Apply changes to files as per guidelines
+
+Changed Files:
+- README.md
+- CONTRIBUTING.md
+
+Reasons for Changes:
+- This is an example message, but anyways: The guidelines changed, so this
+  commit adapts the documents to follow the new guidelines as of 2018-03-14.
+```
+
+
+## Document Design Guidelines
+
+Contributions to documents for the repository should follow these guidelines:
+
+- Lines should be no more than 80 characters
```

Comment:
Why? It makes the document harder to edit, imo. All text editors support text wrapping.

Reply:
Vim hard-wraps (at least in my case) all Git commits, and as discussed for the sole purpose of this pull-request, GitHub *sucks* at showing the context of a line. If someone types out a whole paragraph, sure, GitHub wraps it. However, you'll end up saying something along the lines of "four sentences in, there's a typo" rather than just commenting on one 80-character line.

Reply:
I second the 80 characters limit. Otherwise diffs get difficult to review.

Reply:
BTW, `.editorconfig` would be nice.